### PR TITLE
Support for PHP 8.2

### PR DESCRIPTION
**Pull request by FabienPapet**
wip

### CODE REVIEW

File: classes/ObjectModel.php [hunk 1]
```diff
@@ -26,6 +26,7 @@
 use PrestaShop\PrestaShop\Adapter\ServiceLocator;
 use PrestaShopBundle\Translation\TranslatorComponent;
 
+#[AllowDynamicProperties]
```
Comment (matks) [inline classes/ObjectModel.php:29]: 👀

File: composer.json [hunk 2]
```diff
@@ -126,7 +126,7 @@
         "prestashop/statsstock": "^2",
         "prestashop/translationtools-bundle": "^5.0",
         "sensio/framework-extra-bundle": "^5.1",
-        "smarty/smarty": "^4.1.0",
+        "smarty/smarty": "^4.3.0",
```
Comment (matks) [inline composer.json:129]: Did you check there is no BC breaks between the two? (I know it's not supposed to but... 😅 )

**Review by matks — COMMENTED** [review]
Looks good so far

**Comment by Progi1984** [comment]
It's a duplicate of my PR #29116.

**Comment by FabienPapet** [comment]
Yes @Progi1984, I started to work on it, and for some reason was unable to push my changes to your branch. I was going to make the CI green on this one so you can fetch changes on yours :) 

Sorry if you thought I was stealing your PR  :/

**Comment by Progi1984** [comment]
I need three releases of modules to continue. It is up to date in the header of my PR.

**Pull request closed without merging**